core/src/peer_id.rs

```diff
@@ -24,6 +24,10 @@ use quick_error::quick_error;
 use multihash;
 use std::{convert::TryFrom, fmt, str::FromStr};
 
+/// Public keys with byte-lengths smaller than `MAX_INLINE_KEY_LENGTH` will be
+/// automatically used as the peer id using an identity multihash.
+const MAX_INLINE_KEY_LENGTH: usize = 42;
```

Comment:
Why 42?

Reply:
To match the libp2p go-implementation: https://github.com/libp2p/go-libp2p-core/blob/6e566d10f4a5447317a66d64c7459954b969bdab/peer/peer.go#L34

Reply:
Maybe worth mentioning in a comment.

Reply:
It's part of the libp2p specs. I don't think it's worth adding "this is part of the specs", because it's kind of implied everywhere.

Reply:
> I don't think it's worth adding "this is part of the specs", because it's kind of implied everywhere.

No one suggested that.

Reply:
What would the comment be, then? "To match the Go implementation"? That would be wrong,  it's to match the specs.

Reply:
I was asking "Why 42?" because I did not know where the number came from. The answer I got was "To match the libp2p go-*implementation*" (emphasis mine). Given this I would have considered it useful to state as much in a comment. Then you pointed out that it is specified like this, in other words *not* just implementation defined. Obviously a different answer, so my suggestion no longer applies.

Generally speaking though I would also appreciate comments referring to specs as they can be a great help for the reader.